internal/planner/select.go

```diff
@@ -315,6 +315,19 @@ func findIndexByFilteringField(scanNode *scanNode) immutable.Option[client.Index
 	return immutable.None[client.IndexDescription]()
 }
 
+func findIndexByFieldName(col client.Collection, fieldName string) immutable.Option[client.IndexDescription] {
+	for _, field := range col.Schema().Fields {
+		if field.Name != fieldName {
+			continue
+		}
+		indexes := col.Description().GetIndexesOnField(field.Name)
+		if len(indexes) > 0 {
+			return immutable.Some(indexes[0])
```

Comment:
suggestion: I think I understand why `0` is not a problem and that all we risk is it being potentially less optimised than it could be in the fetcher (e.g. if the `0` index is a composite of 3, and the `1` index is a simple single field index), but I think it is worth a comment here to highlight/confirm/reject this thought :)

Reply:
done